lib/chatter_web/live/chat_live/index.ex

```diff
@@ -68,46 +91,48 @@ defmodule ChatterWeb.ChatLive.Index do
   end
 
   @impl true
-  def handle_event("like", %{"uuid" => uuid, "user" => user} = _params, socket) do
-    MessageAgent.set_like(MessageAgent, {uuid, user})
+  def handle_event("like", %{"id" => id, "user" => user} = _params, socket) do
+    MessageAgent.set_like(MessageAgent, {String.to_integer(id), user})
 
-    {:noreply, assign_search_messages(socket)}
+    {:noreply, socket}
   end
 
   def assign_default(socket) do
     socket
     |> assign(show_write: false)
     |> assign(show_menu: false)
-    |> assign(messages: MessageAgent.get())
     |> assign(changeset: Message.change_message(%Message{}))
     |> assign(username: elem(Generator.run(), 1))
-    |> assign(search: Search.change_search(%Search{}))
+    |> assign(:search, Search.change_search(%Search{}))
     |> assign(filter_option: nil)
   end
 
-  def assign_search_messages(socket) do
-    assign(socket,
-      messages:
-        socket.assigns.search
-        |> apply_changes
-        |> get_filtered_messages
-    )
+  def assign_messages(socket) do
+    stream(socket, :messages, MessageAgent.get())
   end
 
-  def assign_filter_option_messages(socket) when socket.assigns.filter_option != nil do
-    assign(socket, messages: filter_messages_by_option(socket.assigns.filter_option))
-  end
+  defp created_response({message, socket}) do
+    search = apply_changes(socket.assigns.search)
+    is_member =
+      {search, socket.assigns.filter_option}
+      |> MessageFilter.filter_by_params()
+      |> Enum.member?(message)
 
-  def assign_filter_option_messages(socket) do
-    assign_search_messages(socket)
+    {is_member, message, socket} |> created_member_response()
   end
 
+  defp created_member_response({false, _message, socket}), do: {:noreply, socket}
+
+  defp created_member_response({true, message, socket}), do: {:noreply, stream_insert(socket, :messages, message, at: 0)}
```

Comment:
returning `:noreply` within a function you restrict use of composition

Reply:
`:noreply` moved to `handle_info` returns